### PR TITLE
Added CellEditing samples,data and templates

### PR DESCRIPTION
**Pull request by tishko0**

### CODE REVIEW

**Comment by IMinchev64** [comment]
I am currently playing around with the generated React sample from this configuration and several things seem to not work.

Some errors are coming from the `webGridCellEditCellTemplate`:
- the call to `this.grid1` as there is no `grid1` generated but rather `hierarchicalGrid1` in the sample.
- use of attribute `textContext` in `IgrSelectItem` gives:
```
Type '{ selected: boolean; textContent: any; value: any; key: string; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<IgrSelectItem> & Readonly<IIgrSelectItemProps>'.
  Property 'textContent' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<IgrSelectItem> & Readonly<IIgrSelectItemProps>'.
```

- use of `size` property in `IgrSelect` gives:
```
Type '{ children: any; size: string; key: string; change: (x: any) => void; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<IgrSelect> & Readonly<IIgrSelectProps>'.
  Property 'size' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<IgrSelect> & Readonly<IIgrSelectProps>'.
```

Finally, the row islands in the sample show no data for some reason.

I am using igniteui-react version 18.6.0 for testing.

**Comment by tishko0** [comment]
Not really sure where size and textContext were coming from, they were not in the templates so, but  the templates were used in the flat grid as well, so might have been causing it. I added new templates so it should be running okay now.

File: samples/grids/hierarchical-grid/editing-events.json [hunk 1]
```diff
@@ -0,0 +1,165 @@
+{
+    "skipAlterDataCasing": true,
+    "descriptions": {
+      "content": {
+        "type": "WebHierarchicalGrid",
+        "dataRef": "SingersData",
+        "autoGenerate": false,
+        "cellEditRef": "WebGridEditingEventsCellEdit",
```
Comment (IMinchev64) [inline samples/grids/hierarchical-grid/editing-events.json]: The `WebGridEditingEventsCellEdit` template does not correspond to `SingersData`.

**Review by IMinchev64 — CHANGES_REQUESTED** [review]
When you generate the cell-editing-styling samples for WC, React, and Blazor the custom styling is generated but not applied.

**Comment by tishko0** [comment]
> When you generate the cell-editing-styling samples for WC, React, and Blazor the custom styling is generated but not applied.

added id to grid so styles should display properly now

File: samples/grids/hierarchical-grid/editing-events.json (hunk 1 above)
Comment (tishko0) [reply samples/grids/hierarchical-grid/editing-events.json]: fixed the data, but the cellEditRef property is not property of rowIslands so I have it added on the main grid for now only.

**Review by IMinchev64 — APPROVED** [review]

**Pull request merged**